lte/cloud/go/services/lte/analytics/calculations/state.go

```diff
@@ -149,24 +146,39 @@ func (x *UserMetricsCalculation) Calculate(prometheusClient query_api.Prometheus
 		}
 		users := make(map[string]struct{})
 		var exists = struct{}{}
-		var activeSessionsPerGateway map[string]float64
-		var activeSessionsPerAPN map[string]float64
+		activeSessionsPerAPN := make(map[string]float64)
 		for stateID, st := range states {
-			reportedSt := st.ReportedState.(*map[string]string)
-			sessionState, _ := (*reportedSt)["lifecycle_state"]
-			deviceID, _ := (*reportedSt)["device_id"]
-			apnID, _ := (*reportedSt)["apn"]
-			users[stateID.DeviceID] = exists
-			if sessionState == "SESSION_ACTIVE" && deviceID != "" && apnID != "" {
-				activeSessionsPerGateway[deviceID]++
-				activeSessionsPerAPN[apnID]++
+			if st.ReportedState != nil {
+				reportedSt, ok := st.ReportedState.(*state.ArbitraryJSON)
+				if !ok {
+					glog.Infof("error: expected ArbitraryJSON type got %T", st.ReportedState)
```

Comment:
Add more detail. If this pops up as-is in the error logs we're not gonna know what it means. Also, use `glog.Errorf` rather than `"error: ..."`